### DISCUSSION TITLE
nx release in independent mode asks for version bump on unchanged projects

### DISCUSSION THREAD

**Discussion by herrmannben** [nrwl/nx#23390] [Feature Requests]
### Current Behavior

Given an integrated repo with two libraries (`lib-a` and `lib-b`) and one app (`app-a`)
`lib-a` is a dependency of `app-a` .
After a change in `lib-a`, when running `nx release version`, Nx asks for version bump for `app-a`, `lib-a` and `lib-b`

### Expected Behavior

Nx should ask for version bump only for `lib-a` (the updated lib) and `app-a` (the app using `lib-a`) but not for `lib-b`.

### GitHub Repo

https://github.com/herrmannben/test-nx-release

### Steps to Reproduce

1. Clone the repo
2. Do a change in `lib-a` (Change the return value of the function)
3. Commit
4. Run `nx release --skip-publish`

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 20.9.0
   OS     : win32-x64
   npm    : 10.1.0

   nx (global)        : 18.0.4
   nx                 : 18.0.4
   @nx/js             : 18.0.4
   @nx/jest           : 18.0.4
   @nx/linter         : 18.0.4
   @nx/eslint         : 18.0.4
   @nx/workspace      : 18.0.4
   @nx/devkit         : 18.0.4
   @nx/esbuild        : 18.0.4
   @nx/eslint-plugin  : 18.0.4
   @nx/node           : 18.0.4
   @nrwl/tao          : 18.0.4
   typescript         : 5.3.3
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by origranot** [comment]
Same here, any update?
@fahslaj @JamesHenry

**Comment by fahslaj** [comment]
Hi @herrmannben and @origranot, thank you for opening this issue. I'm not sure yet if this should be the default behavior, but it makes sense that you would want to automatically skip over the unchanged packages in a workflow where independent packages are released as-needed and not all together.

For now, you can put together a workaround using `nx release --projects="app-a,lib-a"`. This will only run the release for the specified projects. With some scripting magic, I think you could get close to the desired behavior by combining this `--projects` filter with `nx show projects --affected` to return a list of changed projects.

**Comment by herrmannben** [comment]
Thank you for your response.
For now, the `nx release --projects="app-a,lib-a"` is less relevant as we have a lot of libraries and services in the monorepo, and wanted a safe way using NX to bump versions of services and libraries that don't have a change but use a dependency that has been updated.
We also tried to write a script using `nx show projects --affected` but we were getting all the packages in the list if the package-lock file has changed.

**Comment by or-orchid** [comment]
+1 for this request

**Comment by therockstorm** [comment]
Related issue: https://github.com/nrwl/nx/issues/27887.
